tests/org.palladiosimulator.editors.sirius.tests/src/org/palladiosimulator/editors/sirius/tests/AnchorProviderTest.java

```diff
@@ -0,0 +1,695 @@
+            this.createEdge();
+            this.initEdgeEndPoints();
+            if (compareWithEdgeEndPoints) {
+                this.compareAnchorWithEdgeEndPoints();
+            }
+        }
+
+        private void initElements(final String srcName, final String targetName) {
+            // Source Node
+            this.srcNode = AnchorProviderTest.this.getDiagramElementsFromLabel(AnchorProviderTest.this.diagram, srcName)
+                .get(0);
+            this.srcNodeEP = (NodeEditPart) AnchorProviderTest.this.getEditPart(this.srcNode);
+            this.srcNodeRect = AnchorProviderTest.this.getEditPart(this.srcNode)
+                .getFigure()
+                .getBounds();
+
+            // Target Node
+            this.targetNode = AnchorProviderTest.this
+                .getDiagramElementsFromLabel(AnchorProviderTest.this.diagram, targetName)
+                .get(0);
+            this.targetNodeEP = (NodeEditPart) AnchorProviderTest.this.getEditPart(this.targetNode);
+            this.targetNodeRect = AnchorProviderTest.this.getEditPart(this.targetNode)
+                .getFigure()
+                .getBounds();
+        }
+
+        private void initAnchorPoints() {
+            final Point referencePointForSource = this.srcNodeEP
+                .getSourceConnectionAnchor(AnchorProviderTest.this.cursor)
+                .getLocation(new PrecisionPoint(0, 0));
+            final Point referencePointForTarget = this.targetNodeEP
+                .getTargetConnectionAnchor(AnchorProviderTest.this.cursor)
+                .getLocation(new PrecisionPoint(0, 0));
+            this.targetAnchorPoint = this.targetNodeEP.getTargetConnectionAnchor(AnchorProviderTest.this.cursor)
+                .getLocation(referencePointForSource);
+            this.srcAnchorPoint = this.srcNodeEP.getSourceConnectionAnchor(AnchorProviderTest.this.cursor)
+                .getLocation(referencePointForTarget);
+        }
+
+        private void createEdge() {
+            assertEquals(0, AnchorProviderTest.this.diagram.getEdges()
+                .size());
+            AnchorProviderTest.this.applyEdgeCreationTool(this.nameOfEdgeCreationTool, AnchorProviderTest.this.diagram,
+                    (EdgeTarget) this.srcNode, (EdgeTarget) this.targetNode);
+            this.dEdge = AnchorProviderTest.this.diagram.getEdges()
+                .get(0);
+            assertNotNull(this.dEdge);
+            this.connection = ((ConnectionEditPart) AnchorProviderTest.this.getEditPart(this.dEdge))
+                .getConnectionFigure();
+        }
+
+        private void initEdgeEndPoints() {
+            this.srcEdgeEndPoint = this.connection.getPoints()
+                .getFirstPoint();
+            this.targetEdgeEndPoint = this.connection.getPoints()
+                .getLastPoint();
+        }
+
+        private void compareAnchorWithEdgeEndPoints() {
+            assertEquals(
+                    "Source of " + this.nameOfEdgeCreationTool
```

Comment:
To avoid copying, the strings (`"Source of "`, `"Target of "`, `": The Edges Endpoint differs from the Anchor Point location"`, etc.) could be kept as constants or moved to a further helper method.